src/main/java/model/People.java

```diff
@@ -18,16 +23,50 @@ private void validatePersonCount(final int personCount) {
         }
     }
 
+    private void validateDuplicatedPersonNames(final List<Person> personGroup) {
+        final long personCount = personGroup.stream()
+                .map(Person::getPersonName)
+                .distinct()
+                .count();
+        if (personCount != personGroup.size()) {
+            throw new IllegalArgumentException("중복된 사람이름은 존재할 수 없습니다.");
+        }
+    }
+
     public static People from(final List<String> names) {
-        final List<Person> personGroup = names.stream()
-                .map(Person::new)
-                .toList();
+        final List<Person> personGroup = new ArrayList<>(names.size());
+        for (int column = 0; column < names.size(); column++) {
+            final Person person = Person.from(names.get(column), column);
+            personGroup.add(person);
+        }
         return new People(personGroup);
     }
 
+    public PresentMatches climbDown(final Ladder ladder, final Presents presents) {
+        personGroup.forEach(person -> person.climbDown(ladder));
+        Map<PersonName, Present> personNamePresent = personGroup.stream()
+                .collect(Collectors.toMap(
+                        Person::getPersonName,
+                        person -> presents.getPresent(person.getColumn())));
+        return new PresentMatches(personNamePresent);
```

Comment:
Stream을 이용한 일종의 잡(?)기술을 활용하면 아래와 같이 표현할 수도 있습니다.


```suggestion
        return personGroup.stream()
                .collect(Collectors.collectingAndThen(
                        Collectors.toMap(
                                Person::getPersonName,
                                person -> presents.getPresent(person.getColumn())
                        ),
                        PresentMatches::new)
                );
```

Reply:
1단계 리뷰에서 알게되었던 메서드인데 역시 슥 보고 지나친 메서드라 기억하지 못했었네요 😂😂😂
사용해보겠습니다!